dashboard/src/actions/authActions.js

```diff
@@ -7,6 +7,49 @@ import Cookies from "js-cookie";
 import { showToast } from "actions/toastActions";
 import { uid } from "../utils/helper";
 
+
+// Create an Authentication Request
+export const authenticationRequest = () => async (dispatch, getState) => {
+    try {
+      const endpoints = getState().apiEndpoint.endpoints;
+      const oidcServer = endpoints["openid-connect"]?.issuer;
+      const oidcRealm = endpoints["openid-connect"]?.realm;
+      const oidcClient = endpoints["openid-connect"]?.client;
+      // URI parameters ref: https://openid.net/specs/openid-connect-core-1_0.html#AuthorizationEndpoint
+      // Refer Step 3 of pbench/docs/user_authentication/third_party_token_management.md
+      let req = oidcServer + '/realms/' + oidcRealm + '/protocol/openid-connect/auth';
+      req += '?client_id=' + oidcClient;
+      req += '&response_type=code';
+      req += '&redirect_uri=' + window.location.href.split('?')[0];
+      req += '&scope=profile';
+      req += '&prompt=login';
+      req += '&max_age=120';
+      window.location.href = req;
+    } catch (error) {
+      const alerts = getState().userAuth.alerts;
+      let alert = {};
+      if (error?.response) {
+        alert = {
+          title: error?.response?.data?.message,
+          key: uid(),
+        };
+        dispatch(toggleLoginBtn(true));
+      } else {
+        alert = {
+          title: error?.message,
+          key: uid(),
+        };
+        dispatch({ type: TYPES.OPENID_ERROR });
+      }
+      alerts.push(alert);
+      dispatch({
+        type: TYPES.USER_NOTION_ALERTS,
+        payload: alerts,
+      });
+      dispatch({ type: TYPES.COMPLETED });
```

Comment:
Alternately,
```
      dispatch(error?.response
        ? toggleLoginBtn(true)
        : { type: TYPES.OPENID_ERROR }
      );

      const alert = {
          title: error?.response ? error.response.data?.message : error?.message,
          key: uid(),
      };

      dispatch({
        type: TYPES.USER_NOTION_ALERTS,
        payload: [alerts..., alert],
      });

      dispatch({ type: TYPES.COMPLETED });
```
I think that this does the same thing as your proposed code, with the correction for mutating the Redux state and somewhat tighter focus.